src/evalfunc.c

```diff
@@ -11037,8 +11119,37 @@ f_sqrt(typval_T *argvars, typval_T *rettv)
     else
 	rettv->vval.v_float = 0.0;
 }
+#endif
 
 /*
+ * "rand()" function
+ */
+    static void
+f_srand(typval_T *argvars, typval_T *rettv)
+{
+    if (rettv_list_alloc(rettv) == FAIL)
+	return;
+    if (argvars[0].v_type == VAR_UNKNOWN)
+	list_append_number(rettv->vval.v_list, 123456789);
+    else if (argvars[0].v_type == VAR_NUMBER)
```

Comment:
This should use `get_tv_number_chk`, for Vim functions it is expected to automatically convert strings to numbers.